iphone/Classes/TiApp.m

```diff
@@ -780,30 +780,61 @@ - (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task didSend
   [[NSNotificationCenter defaultCenter] postNotificationName:kTiURLUploadProgress object:self userInfo:dict];
 }
 
-- (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task didCompleteWithError:(NSError *)error
+-(void)URLSession:(NSURLSession *)session dataTask:(NSURLSessionDataTask *)dataTask didReceiveData:(NSData *)data
 {
-  NSMutableDictionary *dict = [NSMutableDictionary dictionaryWithObjectsAndKeys:
-                                                       [NSNumber numberWithUnsignedInteger:task.taskIdentifier], @"taskIdentifier",
-                                                   nil];
+    if(!uploadTaskResponses){
+        uploadTaskResponses = [[NSMutableDictionary alloc] init];
+    }
+    NSMutableDictionary *responseObj =  [uploadTaskResponses objectForKey:@(dataTask.taskIdentifier)];
+    if (!responseObj) {
+        NSMutableData * responseData = [NSMutableData dataWithData:data];
```

Comment:
Why do you need a mutable copy here?

Reply:
I am not sure if I can modify the dictionary key itself. Which is necessary if the delegate method is called more than once and data needs to be appended.